src/Perpetuum/Zones/Terrains/Layer.cs

```diff
@@ -53,6 +53,13 @@ public Layer(LayerType layerType,T[] rawData, int width, int height) : base(laye
 
         public unsafe void CopyFromStreamToArea(Stream stream, Area area)
         {
+            if (area.Width > Width || area.Height > Height)
```

Comment:
So if any part of the update is out of bounds then the whole update fails? Wouldn't this limit the ability to do work close to edges? Would it be better to accept everything that is within the area?

Reply:
That would require trimming that data from the stream, which is a little trickier.  We can make that an enhancement, but it requires a little more precision and understanding how the data is ordered and structured in that packet.  1st pass right now is preventing the app from crashing when design team is terraforming new islands.